jedi/api/environment.py

```diff
@@ -46,49 +44,46 @@ def _sha256(self):
             return self._hash
 
 
+def _get_info():
+    return (
+        sys.executable,
+        sys.prefix,
+        sys.version_info[:3],
+    )
+
+
 class Environment(_BaseEnvironment):
     """
     This class is supposed to be created by internal Jedi architecture. You
     should not create it directly. Please use create_environment or the other
     functions instead. It is then returned by that function.
     """
-    def __init__(self, path, executable):
-        self.path = os.path.abspath(path)
+    def __init__(self, executable):
+        try:
+            self._subprocess = get_subprocess(executable)
+            info = self._subprocess._send(None, _get_info)
+        except Exception as exc:
+            raise InvalidPythonEnvironment(
+                "Could not get version information (%r)" % exc)
+
+        self.executable = info[0]
         """
-        The path to an environment, matches ``sys.prefix``.
+        The Python executable, matches ``sys.executable``.
         """
-        self.executable = os.path.abspath(executable)
+        self.path = info[1]
```

Comment:
@davidhalter 
Should we keep `path` here, or remove, or deprecate it instead of `self.prefix`?

Reply:
Keep path for now. I think we should have a discussion about prefix in a separate PR or issue. I probably think we should deprecate it.